### DISCUSSION TITLE
Optionaly change PDF files order for merge.

### DISCUSSION THREAD

**Discussion by ImBigman** [gotenberg/gotenberg#411] [Q&A] [answered]
Is it possible to change the order in which the files are combined from alphabetical to position in request?
```
--form 'files=@"/path/to/pdf3.pdf"' \
--form 'files=@"/path/to/pdf1.pdf"' \
--form 'files=@"/path/to/pdf2.pdf"' \
--form 'files=@"/path/to/pdf4.pdf"' \
```
Or is it only one way: change the filename before?

From docs: 
This route accepts PDF files and merges them alphabetically.
```
curl \
--request POST 'http://localhost:3000/forms/pdfengines/merge' \
--form 'files=@"/path/to/pdf1.pdf"' \
--form 'files=@"/path/to/pdf2.pdf"' \
--form 'files=@"/path/to/pdf3.pdf"' \
--form 'files=@"/path/to/pdf4.pdf"' \
-o my.pdf
```

**Comment by gulien** [comment, marked as answer]
Hello @ImBigman!

The only way is to change the filename **in the request** (i.e., you don't have to change the filename locally, just the filename in the form data). See for instance: https://github.com/gotenberg/gotenberg-php/blob/main/src/MultipartFormDataModule.php#L103.

**Reply by AndreyZakhvatoshin** [reply]
Hello @gulien !
I want to merge pdf files like this:
$files = [];
        foreach ($this->pdf_pages as $name) {
            $files[] = Stream::path(Storage::path($this->dirPath . '/tmp/' . $name));
        }

        $request = Gotenberg::pdfEngines('http://gotenberg:3000')
            ->outputFilename($resultFileName)
            ->merge(...$files);
But the order is confused when file count > 10

**Reply by gulien** [reply]
@AndreyZakhvatoshin should be fixed now => https://github.com/gotenberg/gotenberg-php/releases/tag/v1.0.1

Thanks 😄

**Reply by AndreyZakhvatoshin** [reply]
Thanks! Now it works!

**Comment by cwo-mjaesch** [comment]
The accepted answer didnt help me fully because the linked line Nr doesnt directly show a solution

this is what i could get to work:

curl \
--request POST 'http://localhost:3000/forms/pdfengines/merge' \
--form 'files=@"./realfilename1.pdf";filename="b_2filename.pdf"' \
--form 'files=@"./realfilename2.pdf";filename="a_3filename.pdf"' \
--form 'files=@"./realfilename3.pdf";filename="a_11111111filename.pdf"' \
-o my_changedoutput.pdf

this will output this order:
1. file3
2. file2
3. file1
